### DISCUSSION TITLE
Perceiver IO in Flax

### DISCUSSION THREAD

**Discussion by manifest** [google/flax#1610] [Show and tell]
Hey all,

I've just published [Flax implementation ](https://github.com/manifest/flax-extra/blob/main/src/flax_extra/model/perceiver/_io.py)of the [Perceiver IO](https://arxiv.org/abs/2107.14795) model and all its building blocks as linen modules. The implementation is highly customizable meaning almost any submodule may be easily configured or replaced with a different type.

Examples include:
- [Classification model](https://flax-extra.readthedocs.io/en/latest/notebooks/perceiver_classification/example) pretrained on images from ImageNet.
- [Masked-language model](https://flax-extra.readthedocs.io/en/latest/notebooks/perceiver_language_modeling/example) pretrained using a large text corpus obtained by combining English Wikipedia and C4.
- [Autoencoder model](https://flax-extra.readthedocs.io/en/latest/notebooks/perceiver_autoencoding/example) pretrained on multimodal input audio, video, and label of the Kinetics-700-2020 dataset.

All the models use pre-trained parameters of the [original DeepMind's models](https://github.com/deepmind/deepmind-research/tree/master/perceiver).
